auto_nag/scripts/component.py

```diff
@@ -0,0 +1,87 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from auto_nag.bugbug_utils import BugbugScript
+from bugbug.models.component import ComponentModel
+
+
+class Component(BugbugScript):
+    def __init__(self):
+        super(Component, self).__init__()
+        self.model = ComponentModel.load(self.retrieve_model('component'))
+
+    def description(self):
+        return 'Assing a component to untriaged bugs'
+
+    def name(self):
+        return 'component'
+
+    def template(self):
+        return 'component.html'
+
+    def subject(self):
+        return '[Using ML] Assign a component to untriaged bugs'
+
+    def columns(self):
+        return ['id', 'summary', 'component', 'confidence']
+
+    def get_bz_params(self, date):
+        start_date, end_date = self.get_dates(date)
+        params = {
+            'component': 'Untriaged',
+            'chfield': '[Bug creation]',
+            'chfieldfrom': start_date,
+            'chfieldto': end_date,
+        }
+
+        return params
+
+    def get_data(self):
+        return {'bugs': {}}
+
+    def bughandler(self, bug, data):
+        data['bugs'][bug['id']] = bug
+
+    def get_bugs(self, date='today', bug_ids=[]):
+        # Retrieve bugs to analyze.
+        all_bug_data = super(Component, self).get_bugs(date=date, bug_ids=bug_ids)
+
+        bugs = all_bug_data['bugs']
+
+        # Retrieve history.
```

Comment:
nit: maybe you could retrieve history, comment & attachment in one shot to avoid to have to wait for each of them

Reply:
I'm doing it this way so that in the regression script I can avoid downloading comments and attachments for some bugs, and I wanted to keep as much as possible in common between the two scripts to avoid duplicating code.